substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jfr/Target_jdk_jfr_internal_SecuritySupport.java

```diff
@@ -27,27 +27,28 @@
 import java.util.List;
 
 import com.oracle.svm.core.annotate.Alias;
-import com.oracle.svm.core.annotate.RecomputeFieldValue;
 import com.oracle.svm.core.annotate.Substitute;
 import com.oracle.svm.core.annotate.TargetClass;
+import com.oracle.svm.core.jdk.JDK21OrEarlier;
 import com.oracle.svm.core.util.VMError;
 
-import jdk.jfr.internal.SecuritySupport.SafePath;
-
-@TargetClass(value = jdk.jfr.internal.SecuritySupport.class, onlyWith = HasJfrSupport.class)
+@TargetClass(value = jdk.jfr.internal.SecuritySupport.class, onlyWith = {HasJfrSupport.class, JDK21OrEarlier.class})
+@SuppressWarnings("unused")
 public final class Target_jdk_jfr_internal_SecuritySupport {
-    // Checkstyle: stop
-    @Alias @RecomputeFieldValue(kind = RecomputeFieldValue.Kind.Reset) //
-    static SafePath JFC_DIRECTORY;
-    @Alias @RecomputeFieldValue(kind = RecomputeFieldValue.Kind.Reset) //
-    static SafePath JAVA_IO_TMPDIR;
-    // Checkstyle: resume
 
     @Substitute
-    public static List<SafePath> getPredefinedJFCFiles() {
+    public static List<Target_jdk_jfr_internal_SecuritySupport_SafePath> getPredefinedJFCFiles() {
         throw VMError.shouldNotReachHere("Paths from the image build must not be embedded into the Native Image.");
     }
 
     @Alias
-    static native SafePath getPathInProperty(String prop, String subPath);
+    static native Target_jdk_jfr_internal_SecuritySupport_SafePath getPathInProperty(String prop, String subPath);
```

Comment:
I think that `SubstrateJVM.getDumpPath()` needs to now accomodate the case where we're on `JDKLatest`. Right now it still unconditionally uses `Target_jdk_jfr_internal_SecuritySupport.getPathInProperty` which won't exist on `JDKLatest`

Reply:
thanks @roberttoyonaga! 

@christianhaeubl do you know the answer?

Reply:
@roberttoyonaga : good point, I pushed a fix.